src/subapps/dataExplorer/ProjectSelector.tsx

```diff
@@ -0,0 +1,86 @@
+import { ProjectResponseCommon } from '@bbp/nexus-sdk';
+import { useNexusContext } from '@bbp/react-nexus';
+import { AutoComplete, Input, notification } from 'antd';
+import React, { useState } from 'react';
+import { useQuery } from 'react-query';
+import './styles.less';
+
+interface Props {
+  onSelect: (orgLabel?: string, projectLabel?: string) => void;
+}
+
+export const ProjectSelector: React.FC<Props> = ({ onSelect }: Props) => {
+  const { data: projects } = useProjects();
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const allOptions = [
+    { value: AllProjects, label: AllProjects },
+    ...(projects?.map(projectToOption) ?? []),
+  ].filter(project => project.value.includes(searchTerm));
+
+  return (
+    <div className="search-container">
+      <span className="label">Show me </span>
+      <AutoComplete
+        options={allOptions}
+        style={{ width: 200 }}
+        defaultValue={AllProjects}
+        onSearch={text => {
+          setSearchTerm(normalizeSearchTerm(text));
+        }}
+        onSelect={text => {
+          if (text === AllProjects) {
+            onSelect(undefined, undefined);
+          } else {
+            const [org, project] = text.split('/');
+            onSelect(org, project);
+          }
+        }}
+        aria-label="project-filter"
+        bordered={false}
+        className="search-input"
+        popupClassName="search-menu"
+      >
+        <Input.Search size="middle" />
+      </AutoComplete>
+    </div>
+  );
+};
+
+export const AllProjects = 'All projects';
+
+const projectToOption = (
+  project: ProjectResponseCommon
+): { value: string; label: string } => {
+  return {
+    value: `${project._organizationLabel}/${project._label}`,
+    label: `${project._organizationLabel}/${project._label}`,
+  };
+};
+
+const useProjects = () => {
+  const nexus = useNexusContext();
+  return useQuery({
+    queryKey: ['data-explorer-projects'],
+    retry: false,
+    queryFn: async () => {
+      // TODO: Replace this with aggregation API when it is ready.
+      // NOTE: At the moment it is not possible to get all projects a user has access to in 1 request. To get around this, we make 2 requests:
+      // 1st request -> Get the total number of projects (n) a user can see
+      // 2nd request -> Use the total retrieved from above request to specify the size of the projects to return.
+      const firstPageOfProjects = await nexus.Project.list(undefined, {
+        size: 1,
+      });
```

Comment:
I think the total projects are not high, so i suggest to use predefined value  as 100, without using a handshake request just to get the total and as you said in the comment when the api will be ready we can have a clean way to do it

Reply:
If it's not a big deal, I would like to keep it as it is atm.

The number of projects in prod is 138 (so more than 100). The solution right now will work in all the cases, as it is, and will always show the *correct* result (at the cost of 1 extra api call that took 259 ms on VPN). Like you said, this will soon be replaced by aggregation api.